### DISCUSSION TITLE
upgrade to kong 2.7.1 after the pressure test performance is not as good as kong 2.5.1

### DISCUSSION THREAD

**Discussion by chuan-T** [Kong/kong#8605] [Help]
I deployed different versions of kong using the same resource limits(1 CPU), the same backend, the same request, without any plugins in kong proxy

in version 2.5.1, my pressure tests performed as follows
<img width="1436" alt="image" src="https://user-images.githubusercontent.com/43464580/160546292-9edeac0d-9edf-425d-b63c-b780b8673101.png">

in version 2.7.1, my pressure tests performed as follows
<img width="1438" alt="image" src="https://user-images.githubusercontent.com/43464580/160546451-4e88d4bd-eea4-4533-b55b-14fccac34074.png">

in my pressure tests, 2.7.1`s performance is worse than 2.5.1
## problem
Is this performance issue expected? I see that the release log contains changes related to performance optimization

**Comment by mayocream** [comment]
Hi @chuan-T, Would you like to share the steps of your performance test with us?

**Reply by chuan-T** [reply]
sure
1. upgrade one of my test environments from 2.5.1 to 2.7.1, use the same custom nginx template and resource In my k8s cluster
2. deploy a test backend,  This service only prints my request parameters and returns them directly. The performance bottleneck is not here
3. configure a transparent proxy at kong
4. use wrk in another container in my k8s cluster to request the kong's ip address directly
5. the performance after separate requests is like the above 
@mayocream

**Reply by mayocream** [reply]
@chuan-T If the performance test results are easy to reproduce in a clean install, please create a GitHub issue for tracking:

Please make sure the issue includes the following information:
- Setup steps and configurations.
- Simple test scripts.
- Test env infomration.
- Test resutls.

**Comment by hanshuebner** [comment]
Hello @chuan-T,

thank you for reporting the change in performance behaviour that you have observed between Kong 2.5.1 and Kong 2.7.1.  We are unable to reproduce your results, however.  I have run tests against versions 2.5.1, 2.7.1 and master and if anything can be observed, it is that Kong got a little bit faster with each version (see below for results).  For my tests, I used Docker-based setups on my MacBook Pro using a trivial NodeJS based server as the backend that is hit by Kong.

Given these results, I wonder if you're able to reproduce your reported results, and if you can supply more detailed information about the setup that you're running your tests in.

Kind regards,
Hans

```
# Kong 2.5.1
pollux% http :60080 | jq -r .version
2.5.1
pollux% wrk -t 10 -c 100 -d 5m --latency --timeout 5s http://localhost:60079/foo
Running 5m test @ http://localhost:60079/foo
  10 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    13.53ms    9.41ms 140.36ms   85.79%
    Req/Sec   806.07    134.66     1.33k    69.37%
  Latency Distribution
     50%   10.60ms
     75%   16.07ms
     90%   25.07ms
     99%   50.01ms
  2407024 requests in 5.00m, 486.65MB read
Requests/sec:   8020.91
Transfer/sec:      1.62MB
# Kong 2.7.1
pollux% http :59785 | jq -r .version
2.7.1
pollux% wrk -t 10 -c 100 -d 5m --latency --timeout 5s http://localhost:59788/foo
Running 5m test @ http://localhost:59788/foo
  10 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    13.27ms    9.43ms 162.77ms   86.28%
    Req/Sec   826.47    132.48     1.29k    67.87%
  Latency Distribution
     50%   10.26ms
     75%   15.70ms
     90%   24.85ms
     99%   49.92ms
  2467696 requests in 5.00m, 505.57MB read
Requests/sec:   8223.43
Transfer/sec:      1.68MB
# master
pollux% http :59793 | jq -r .version
2.8.0
pollux% wrk -t 10 -c 100 -d 5m --latency --timeout 5s http://localhost:59792/foo
Running 5m test @ http://localhost:59792/foo
  10 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    12.92ms    9.03ms 116.30ms   85.06%
    Req/Sec   844.38    137.38     1.67k    69.50%
  Latency Distribution
     50%   10.03ms
     75%   15.62ms
     90%   24.30ms
     99%   46.99ms
  2521268 requests in 5.00m, 483.55MB read
Requests/sec:   8401.41
Transfer/sec:      1.61MB
```
